### PR TITLE
Resolve #443 #445  -- Some CCs

### LINKED ISSUE #443: Invulnerability
Invulnerability

### PR DESCRIPTION
**Pull request by Deudly**
Invulnerability CC
Disarm CC

Refs #443 #445

### CODE REVIEW

File: GameServerLib/Logic/GameObjects/Unit.cs [hunk 1]
```diff
@@ -450,7 +450,11 @@ public virtual void DealDamageTo(Unit target, float damage, DamageType type, Dam
 
             //Damage dealing. (based on leagueoflegends' wikia)
             damage = defense >= 0 ? (100 / (100 + defense)) * damage : (2 - (100 / (100 - defense))) * damage;
-
+            if (target.HasCrowdControl(CrowdControlType.Invulnerable))
```
Comment (molenzwiebel) [inline GameServerLib/Logic/GameObjects/Unit.cs]: This does not account for pure damage, which should bypass invulnerability.

**Review by molenzwiebel — CHANGES_REQUESTED** [review]

File: GameServerLib/Logic/GameObjects/Unit.cs (hunk 1 above)
Comment (Deudly) [reply GameServerLib/Logic/GameObjects/Unit.cs]: @molenzwiebel @MythicManiac Pure damage? What's that? You mean true damage?

Comment (Deudly) [reply GameServerLib/Logic/GameObjects/Unit.cs]: Btw, Tryndamere Ult blocks all damage types I think

**Review by MythicManiac — APPROVED** [review]
Seems good

**Pull request merged**